XenAdminTests/HealthCheckTests/CredentialTests.cs

```diff
@@ -58,20 +58,17 @@ public void CredentialOp()
             string HostName = "Host1";
             string UserName = "User1";
             string Password = "password1";
-            int conSize = ServerListHelper.instance.GetServerList().Count;
-            //1. Empty credential 
-            NamedPipeClientStream pipeClient = new NamedPipeClientStream(".", HealthCheckSettings.HEALTH_CHECK_PIPE, PipeDirection.Out);
-            pipeClient.Connect();
-            string credential = EncryptionUtils.ProtectForLocalMachine(String.Join(SEPARATOR.ToString(), new[] { HostName, null, null }));
-            pipeClient.Write(Encoding.UTF8.GetBytes(credential), 0, credential.Length);
-            pipeClient.Close();
+

```

Comment:
As discussed, I think we may want to keep the original testcase and additionally ensure beforehand that the list is empty. Also, we should add the sleep before the assertion.
Now, we may want to reconsider the way we handle these nulls in the code, because the current behaviour is a bit strange. Perhaps ignore them, if we don't want to explicitly remove them.